compiler/src/iree/compiler/Codegen/Utils/GPUUtils.cpp

```diff
@@ -967,12 +967,25 @@ IREE::GPU::TargetAttr getCLGPUTarget(MLIRContext *context) {
   return IREE::GPU::getFullTarget(backend, arch, features, context);
 }
 
-IREE::GPU::TargetAttr getGPUTargetAttr(IREE::HAL::ExecutableTargetAttr target) {
-  if (auto config = target.getConfiguration()) {
-    if (auto attr = config.getAs<IREE::GPU::TargetAttr>(kGPUTargetAttrName))
-      return attr;
+IREE::GPU::TargetAttr getGPUTargetAttr(Attribute attr) {
+  if (!attr) {
+    return getCLGPUTarget(attr.getContext());
   }
```

Comment:
wait, it'd crash, right? If it is a null, you can't access `.getContext()`? I think we just return `nullptr` in this case.

Reply:
Thanks for the catch, changed it.